exporter/awskinesisexporter/factory.go

```diff
@@ -45,59 +53,49 @@ func createDefaultConfig() config.Exporter {
 		AWS: AWSConfig{
 			Region: "us-west-2",
 		},
-		KPL: KPLConfig{
-			BatchSize:            5242880,
-			BatchCount:           1000,
-			BacklogCount:         2000,
-			FlushIntervalSeconds: 5,
-			MaxConnections:       24,
-		},
-
-		QueueSize:            100000,
-		NumWorkers:           8,
-		FlushIntervalSeconds: 5,
-		MaxBytesPerBatch:     100000,
-		MaxBytesPerSpan:      900000,
+		MaxRecordsPerBatch: batch.MaxBatchedRecords,
+		MaxRecordSize:      batch.MaxRecordSize,
 	}
 }
 
-func createTracesExporter(
-	_ context.Context,
-	params component.ExporterCreateSettings,
-	config config.Exporter,
-) (component.TracesExporter, error) {
-	c := config.(*Config)
-	k, err := awskinesis.NewExporter(&awskinesis.Options{
-		Name:               c.ID().String(),
-		StreamName:         c.AWS.StreamName,
-		AWSRegion:          c.AWS.Region,
-		AWSRole:            c.AWS.Role,
-		AWSKinesisEndpoint: c.AWS.KinesisEndpoint,
-
-		KPLAggregateBatchSize:   c.KPL.AggregateBatchSize,
-		KPLAggregateBatchCount:  c.KPL.AggregateBatchCount,
-		KPLBatchSize:            c.KPL.BatchSize,
-		KPLBatchCount:           c.KPL.BatchCount,
-		KPLBacklogCount:         c.KPL.BacklogCount,
-		KPLFlushIntervalSeconds: c.KPL.FlushIntervalSeconds,
-		KPLMaxConnections:       c.KPL.MaxConnections,
-		KPLMaxRetries:           c.KPL.MaxRetries,
-		KPLMaxBackoffSeconds:    c.KPL.MaxBackoffSeconds,
-
-		QueueSize:             c.QueueSize,
-		NumWorkers:            c.NumWorkers,
-		MaxAllowedSizePerSpan: c.MaxBytesPerSpan,
-		MaxListSize:           c.MaxBytesPerBatch,
-		ListFlushInterval:     c.FlushIntervalSeconds,
-		Encoding:              exportFormat,
-	}, params.Logger)
+func NewTracesExporter(ctx context.Context, params component.ExporterCreateSettings, conf config.Exporter) (component.TracesExporter, error) {
+	return createExporter(ctx, params, conf)
+}
+
+func NewMetricsExporter(ctx context.Context, params component.ExporterCreateSettings, conf config.Exporter) (component.MetricsExporter, error) {
+	return createExporter(ctx, params, conf)
+}
+
+func NewLogsExporter(ctx context.Context, params component.ExporterCreateSettings, conf config.Exporter) (component.LogsExporter, error) {
+	return createExporter(ctx, params, conf)
+}
+
+func createExporter(_ context.Context, params component.ExporterCreateSettings, conf config.Exporter) (*Exporter, error) {
```

Comment:
To align this better with the Kafka exporter, should we consider using more of exporterhelper methods? i.e. for instantiating a new exporter we use `exporterhelper.NewXXXExporter`

Reply:
I am in for that, I need to look into detail what methods they are using and I can carry them over here.

Reply:
iirc it always interfacing with retry and queueing configurations so defo worth checking out. Creating via the exporterhelper also enables the internal metrics for the exporter by default I believe